compiler/qsc_eval/src/val.rs

```diff
@@ -308,7 +338,7 @@ impl Value {
     /// # Panics
     /// This will panic if the [Value] is not a [`Value::Qubit`].
     #[must_use]
-    pub fn unwrap_qubit(self) -> Qubit {
+    pub fn unwrap_qubit(self) -> Weak<Qubit> {
```

Comment:
there is _some_ code golf opportunity in defining a second function, `unwrap_upgraded_qubit()` where you centralize the expect/upgrade calls. It's not unidiomatic because this function name already has `unwrap` in it, denoting that it could panic. So the call to `expect()` would be fine.

Reply:
I pulled this up into the `QubitRef` struct rather than put it directly onto the `Value` impl.